c7n/resources/rds.py

```diff
@@ -107,33 +109,35 @@ def account_id(self):
         return self._account_id
 
     @property
-    def arn_generator(self):
-        if self._arn_generator is None:
-            self._arn_generator = DBInstanceARNGenerator(
-                self.config.region,
-                self.account_id)
-        return self._arn_generator
+    def generate_arn(self):
+        if self._generate_arn is None:
+            self._generate_arn = functools.partial(
+                generate_arn, 'rds', region=self.config.region,
+                account_id=self.account_id, resource_type='db', separator=':')
+        return self._generate_arn
 
     def augment(self, resources):
         filter(None, _rds_tags(
             self.get_model(),
             resources, self.session_factory, self.executor_factory,
-            self.arn_generator))
+            self.generate_arn))
         return resources
 
 
 def _rds_tags(
-        model, dbs, session_factory, executor_factory, arn_generator):
+        model, dbs, session_factory, executor_factory, generate_arn):
     """Augment rds instances with their respective tags."""
 
     def process_tags(db):
         client = local_session(session_factory).client('rds')
-        arn = arn_generator.generate(db[model.id])
+        arn = generate_arn(db[model.id])
+        tag_list = None
         try:
             tag_list = client.list_tags_for_resource(ResourceName=arn)['TagList']
         except ClientError as e:
             if e.response['Error']['Code'] == "DBInstanceNotFound":
                 return None
+            log.warning("Exception getting rds tags  \n %s" % (e))
```

Comment:
so we may need to return none here, so that the instance gets filtered out for this run, there's competing concerns between tag compliance and encryption, the former only wants to consider instances which tags where retrieved for, the later wants to look at the instance regardless of tagging.